test/unit/rules/no-unsupported-role-attributes-test.js

```diff
@@ -0,0 +1,233 @@
+import generateRuleTests from '../../helpers/rule-test-harness.js';
+
+generateRuleTests({
+  name: 'no-unsupported-role-attributes',
+
+  config: true,
+
+  good: [
+    '<div role="button" aria-disabled="true"></div>',
+    '<div role="heading" aria-level="1" />',
+    '<span role="checkbox" aria-checked={{this.checked}}></span>',
+    '<CustomComponent role="banner" />',
+    '<div role="textbox" aria-required={{this.required}} aria-errormessage={{this.error}}></div>',
+    '<dialog />',
+    '<a href="#" aria-describedby=""></a>',
+    '<menu type="toolbar" aria-hidden="true" />',
+    '<menuitem type="command" aria-labelledby={{this.label}} />',
+    '<input type="image" aria-atomic />',
+    '<input type="submit" aria-disabled="true" />',
+    '<select aria-expanded="false" aria-controls="ctrlID" />',
+  ],
```

Comment:
Should have a valid test case to indicate that any random non-aria attribute is allowed:
* `<div role="heading" foo="true" />`
* `<div type="button" foo="true" />`